src/devices/bus/isa/lbaenhancer.cpp

```diff
@@ -0,0 +1,142 @@
+
+Notes:
+
+J3, J2, J1 set BIOS (start) address to C800, CC00, D000, D400, D800 or DC00
+
+- one CD74LS245 
+- one PALCE16V8H-25
+- one 32 K EPROM : 27C256 (STM) (mirrored after 16 K)
+
+*/
+
+#include "emu.h"
+#include "lbaenhancer.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(LBA_GEN, lba_gen_device, "lba_gen", "VS Systems LBA Enhancer BIOS 1995")
+
+//-------------------------------------------------
+//  ROM( lba_gen )
+//-------------------------------------------------
+
+ROM_START( lbabios )
+	ROM_REGION( 0xc8000, "lbabios", 0 )
+	ROM_LOAD( "lbaenhancer.bin", 0x0000, LBA_GEN_ROMSIZE, CRC(39d4566d) SHA1(d275193a870250f212dc29768d4e68fb43770e95) )
+ROM_END
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+
+const tiny_rom_entry *lba_gen_device::device_rom_region() const
+{
+	return ROM_NAME( lbabios );
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - add device configuration
+//-------------------------------------------------
+MACHINE_CONFIG_START( lba_gen_device::device_add_mconfig )
+//	MCFG_DEVICE_ADD("lba_gen", LBA_GEN, 0)
+MACHINE_CONFIG_END
+
+
+//-------------------------------------------------
+//  dip switches  
+//-------------------------------------------------
+static INPUT_PORTS_START( lba_gen_dsw )
+
+	PORT_START("ROM_ADDRESS")
+	PORT_DIPNAME( 0xff, 0x00, "16 K ROM bank (disable shadowing; exclude in EMM386)")
+	PORT_DIPSETTING(    0, "C8000 - CBFFF (sometimes occupied by graphics or hard disk controller)" )
+	PORT_DIPSETTING(    1, "CC000 - CFFFF" ) 
+	PORT_DIPSETTING(    2, "D0000 - D3FFF" ) 
+	PORT_DIPSETTING(    3, "D4000 - D7FFF" ) 
+	PORT_DIPSETTING(    4, "D8000 - DBFFF" ) 
+	PORT_DIPSETTING(    5, "DC000 - DFFFF" ) 
+INPUT_PORTS_END
+
+
+//-------------------------------------------------
+//  isa8_areplay_device - constructor
+//-------------------------------------------------
+ioport_constructor lba_gen_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( lba_gen_dsw );
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+// Byte 3 in extra BIOS specifies 8 blocks of 512 byte (4K)
+void lba_gen_device::device_start()
+{
+	set_isa_device();
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+void lba_gen_device::device_reset()
+{
+	uint32_t current_rom_start = 0xc8000 + (ioport("ROM_ADDRESS")->read()* 0x4000);
+	uint32_t current_rom_end   = current_rom_start + LBA_GEN_ROMSIZE - 1;
+	m_isa->install_rom(this, current_rom_start, current_rom_end,  "lbabios",  "lbabios");
```

Comment:
You need to store a flag to indicate you've installed your handlers and not reinstall them on reset or it leads to trouble since you can't safely unmap the old location.  Changes to address mapping really require a hard reset.